src/UniOneClient.php

```diff
@@ -38,23 +39,21 @@ final class UniOneClient
      */
     private string $apiKey;
 
-    /**
-     * The timeout in seconds.
-     *
-     * @var float
-     */
-    private float $timeout;
-
     /**
      * @param string $endpoint
      * @param string $apiKey
+     * @param array $config
      */
-    public function __construct(string $endpoint, string $apiKey)
+    public function __construct(string $endpoint, string $apiKey, array $config = [])
     {
+        $defaults = [
+            'timeout' => 5,
+            'base_uri' => $this->endpoint,
```

Comment:
`$this->endpoint` will be available only after `$this->setEndpoint($endpoint);` is called.